app/display/model/src/main/java/org/csstudio/display/builder/model/rules/RuleToScript.java

```diff
@@ -276,13 +276,14 @@ else if (pform == PropFormat.FONT)
             for (Map.Entry<String, String> entry : pvm.entrySet())
             {
                 final String varname = entry.getKey();
-                if (expr_to_check.contains(varname))
+                /*
+                 * PVUtil.getDouble() (used for pv0, pv1, etc) does not throw an exception when the PV is disconnected
+                 * Let's make sure that we have a pvInt for every PV
+                 */
+                if (expr_to_check.contains(varname) || varname.contains("pvInt"))

```

Comment:
If the rule only uses `pvStr0` because the data is a string, will adding a `pvInt0` cause problems, since the string cannot be parsed as an integer?

Reply:
I've just tested it and it did not complain. By the way, I was wondering why getDouble() returns NaN when the PV is not connected instead of throwing an exception?

Reply:
OK, thanks for checking.
For a double, NaN is a common representation of "I don't have a number". For an integer, there is no way to express "I don't have a number". Values like -1 or Integer.MIN_VALUE are all potentially valid numbers.

Reply:
Wouldn't it make sense to add a `getDoubleWithException()` variant that would throw an exception instead of returning NaN? Originally rules were not executed when the PV was disconnected so it was impossible to take advantage of checking NaN... and we could get rid of the "mandatory" pvInt

Reply:
getDouble() is meant as a quick way to get a double. Any problem turns into NaN.
If you care about the nature of a specific problem, like PV has no connection, you can already check via PV.isDisconnected.

So maybe instead of changing the PVUtil in all those places, just add a PV.isDisconnected check to the generated rule code.